### PR TITLE
Fixes to massive Networking CLI examples update

### PR DESCRIPTION
**Pull request by markjbrown**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5918
This is an **experimental preview** for @microsoft users.

**Comment by sptramer** [comment]
Please don't submit massive help changes. This makes it harder to copyedit them for consistency, and always seems to necessitate opening a 2nd PR, which doesn't allow me to check that the actual issues were addressed. Keeping changes to go section-by-section is a good choice.

**Comment by markjbrown** [comment]
@sptramer apologies. Will only be massive once. I went line by line through your comments and applied your specific and across-the-board recommendations. --thanks.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 1]
```diff
@@ -1619,28 +2810,54 @@
...
+"""
+
+helps['network route-filter rule list-service-communities'] = """
+    type: command
+    short-summary: Gets all the available BGP service communities.
+    examples:
+        - name: Gets all the available BGP service communities.
+          text: az network route-filter rule list-service-communities -o table
+        - name: Get the community value for Exchange.
+          text: az network route-filter rule list-service-communities --query '[].bgpCommunities[?communityName==`Exchange`].[communityValue][][]' -o tsv
```
Comment (sptramer) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: Split into multiple lines

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 2]
```diff
@@ -1870,250 +3129,379 @@
...
+        For more information on Vnet-to-Vnet connections, visit https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-howto-vnet-vnet-cli
 """
 
 helps['network vpn-connection create'] = """
     type: command
     short-summary: Create a VPN connection.
+    examples:
+        - name: >
+            Create a site-to-site connection between an Azure virtual network and an on-premises local network gateway.
+            NOTE: The VPN Gateway and Local Network Gateway must be provisioned before creating the connection between them.
```
Comment (sptramer) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: Remove `NOTE` and everything after it.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 3]
```diff
@@ -1870,250 +3129,379 @@
...
+          text: |
+            az network vpn-connection create -g MyResourceGroup -n MyConnection --vnet-gateway1 MyVnetGateway \\
+            -local-gateway2 MyLocalGateway --shared-key Abc123
```
Comment (sptramer) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: Add indentation

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 4]
```diff
@@ -1870,250 +3129,379 @@
...
+helps['network vpn-connection ipsec-policy add'] = """
     type: command
-    short-summary: Reset a VPN connection shared key.
+    short-summary: Add a VPN connection IPSec policy.
+    long-summary: Set all IPsec policies of a VPN connection. If you want to set any IPsec policy, you must set them all.
+    examples:
+        - name: Add specified IPsec policies to a connection instead of relying on defaults.
+          text: |
+            az network vpn-connection ipsec-policy add -g MyResourceGroup –-connection-name MyConnection \\
+            -–dh-group DHGroup14 -–ike-encryption AES256 --ike-integrity SHA384 --ipsec-encryption DES3 \\
```
Comment (sptramer) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: Add indentation to continuation lines

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 5]
```diff
@@ -15,260 +15,396 @@
...
     short-summary: List HTTP settings.
+    examples:
+        - name: List HTTP settings.
+          text: az network application-gateway http-settings list -g MyResourceGroup --gateway-name MyAppGateway
 """
 
 helps['network application-gateway http-settings show'] = """
     type: command
-    short-summary: Get the details of a gateway's HTTP settings.
+    short-summary: Get the details of a gateways HTTP settings.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: Done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 6]
```diff
@@ -346,74 +514,125 @@
...
+    examples:
+        - name: Get the details of a rule.
+          text: az network application-gateway rule list -g MyResourceGroup --gateway-name MyAppGateway -n MyRule
 """
 
 helps['network application-gateway rule update'] = """
     type: command
     short-summary: Update a rule.
+    examples:
+        - name: Update a rule use a new Http Listener.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 7]
```diff
@@ -424,83 +643,169 @@
...
+          text: |
+            az network application-gateway ssl-policy set -g MyResourceGroup --gateway-name MyAppGateway \\
+                --policy-type Custom --min-protocol-version TLSv1_2 \\
+                --cipher-suits TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 TLS_RSA_WITH_AES_128_GCM_SHA256
 """
 
 helps['network application-gateway ssl-policy show'] = """
     type: command
-    short-summary: Get the details of a gateway's SSL policy settings.
+    short-summary: Get the details of gateway SSL policy settings.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 8]
```diff
@@ -424,83 +643,169 @@
...
+    examples:
+        - name: Get the details of a URL path map.
+          text: az network application-gateway url-path-map show -g MyResourceGroup --gateway-name MyAppGateway -n MyUrlPathMap
 """
 
 helps['network application-gateway url-path-map update'] = """
     type: command
     short-summary: Update a URL path map.
+    examples:
+        - name: Update a URL path map to use new default http settings.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 9]
```diff
@@ -519,146 +824,32 @@
...
-
 # endregion
 
 # region Application Security Group
 helps['network asg'] = """
     type: group
     short-summary: Manage application security groups (ASGs).
     long-summary: >
-        You can configure network security as a natural extension of an application's structure, ASG allows
+        You can configure network security as a natural extension of an application structure, ASG allows
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 10]
```diff
@@ -667,102 +858,755 @@
...
+    examples:
+        - name: Get details of an application security group.
+          text: az network asg show -g MyResourceGroup -n MyAsg
 """
 
 helps['network asg update'] = """
     type: command
     short-summary: Update an application security group.
+    long-summary: >
+        This command can only be used to update the tags for an application security group. Name and resource group are immutable and cannot be updated.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 11]
```diff
@@ -667,102 +858,755 @@
...
-helps['network dns zone update'] = """
+helps['network dns record-set a remove-record'] = """
     type: command
-    short-summary: Update a DNS zone's properties. Does not modify DNS records within the zone.
-    parameters:
-        - name: --if-match
-          short-summary: Update only if the resource with the same ETAG exists.
+    short-summary: Remove an A record from its record set.
+    long-summary: >
+        By default, if the last record in a set is removed, the record set is deleted. To retain the empty record set, include --keep-empty-record-set.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 12]
```diff
@@ -667,102 +858,755 @@
...
+"""
+
+helps['network dns record-set aaaa remove-record'] = """
+    type: command
+    short-summary: Remove AAAA record from its record set.
+    long-summary: >
+        By default, if the last record in a set is removed, the record set is deleted. To retain the empty record set, include --keep-empty-record-set.
+    examples:
+        - name: Remove an AAAA record from its record set.
+          text: az network dns record-set aaaa remove-record -g MyResourceGroup -z www.mysite.com -n MyRecordSet -a MyIpv6Address
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: Not sure which you wanted broken up so I applied to the longest ones.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 13]
```diff
@@ -667,102 +858,755 @@
...
+    type: group
+    short-summary: Manage DNS CAA records.
+"""
+
+helps['network dns record-set caa add-record'] = """
+    type: command
+    short-summary: Add a CAA record.
+    examples:
+        - name: Add a CAA record.
+          text: az network dns record-set caa add-record -g MyResourceGroup -z www.mysite.com -n MyRecordSet --flags 0 --tag “issue” –-value “ca.contoso.com”
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: same here too, done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 14]
```diff
@@ -667,102 +858,755 @@
...
+"""
+
+helps['network dns record-set caa remove-record'] = """
+    type: command
+    short-summary: Remove a CAA record from its record set.
+    long-summary: >
+        By default, if the last record in a set is removed, the record set is deleted. To retain the empty record set, include --keep-empty-record-set.
+    examples:
+        - name: Remove a CAA record from its record set.
+          text: az network dns record-set caa remove-record -g MyResourceGroup -z www.mysite.com -n MyRecordSet --flags 0 --tag “issue” –-value “ca.contoso.com”
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 15]
```diff
@@ -667,102 +858,755 @@
...
+
+# region DNS MX (Mail) records
+helps['network dns record-set mx'] = """
+    type: group
+    short-summary: Manage DNS MX records.
+"""
+
+helps['network dns record-set mx add-record'] = """
+    type: command
+    short-summary: Add a MX record.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 16]
```diff
@@ -667,102 +858,755 @@
...
+    type: command
+    short-summary: Create an empty CNAME record set.
+    examples:
+        - name: Create an empty CNAME record set.
+          text: az network dns record-set cname create -g MyResourceGroup -z www.mysite.com -n MyRecordSet
+"""
+
+helps['network dns record-set cname delete'] = """
+    type: command
+    short-summary: Delete the CNAME record set and its associated record.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 17]
```diff
@@ -667,102 +858,755 @@
...
+
+# region DNS SRV records
+helps['network dns record-set srv'] = """
+    type: group
+    short-summary: Manage DNS SRV records.
+"""
+
+helps['network dns record-set srv add-record'] = """
+    type: command
+    short-summary: Add a SRV record.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 18]
```diff
@@ -667,102 +858,755 @@
...
+
+# region DNS SOA records
+helps['network dns record-set soa'] = """
+    type: group
+    short-summary: Manage a DNS SOA record.
+"""
+
+helps['network dns record-set soa show'] = """
+    type: command
+    short-summary: Get the details of a SOA record.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 19]
```diff
@@ -913,134 +1834,200 @@
...
           text: >
-            az network lb create -g MyResourceGroup -n MyLb --vnet-name MyVnet --subnet MySubnet
-        - name: Create a zone flavored public facing load balancer through provisioning a zonal public ip.
+            az network lb create -g MyResourceGroup -n MyLb --sku Basic --vnet-name MyVnet --subnet MySubnet
+        - name: Create a basic zone flavored internal load balancer, through provisioning a zonal public ip.
           text: >
-            az network lb create -g MyResourceGroup -n MyLB --public-ip-zone 2
-        - name: Create a zone flavored internal facing load balancer through provisioning a zonal frontend ip configuration.
+            az network lb create -g MyResourceGroup -n MyLb --sku Basic --public-ip-zone 2
+        - name: Create a standard zone flavored public-facing load balancer, through provisioning a zonal frontend ip configuration and Vnet.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 20]
```diff
@@ -913,134 +1834,200 @@
...
 helps['network lb frontend-ip create'] = """
     type: command
     short-summary: Create a frontend IP address.
+    examples:
+        - name: Create a frontend ip address for a public load balancer.
+          text: az network lb frontend-ip create -g MyResourceGroup -n MyFrontendIp --lb-name MyLb --public-ip-name MyFrontendIp
+        - name: Create a frontend ip address for an internal load balancer.
+          text: |
+            az network lb frontend-ip create -g MyResourceGroup -n MyFrontendIp --lb-name MyLb \\
+            --private-ip-address 10.10.10.100 --subnet-name MySubnet --subnet-vnet-name MyVnet
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 21]
```diff
@@ -1122,61 +2143,113 @@
...
-            configuration and port to a backend address pool and port.
+            Create a load balancing rule that assigns a front-facing IP configuration and port to an address pool and port.
           text: |
-            az network lb rule create -g MyResourceGroup --lb-name MyLb -n MyLbRule \\
-                --protocol Tcp --frontend-port 80 --backend-port 80
+            az network lb rule create -g MyResourceGroup --lb-name MyLb -n MyLbRule --protocol Tcp \\
+                --frontend-ip-name MyFrontEndIp --frontend-port 80 \\
+                --backend-pool-name MyAddressPool --backend-port 80
+        - name: >
+            Create a load balancing rule that assigns a front-facing IP configuration and port to an address pool and port with the floating ip feature.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done for all three.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 22]
```diff
@@ -1122,61 +2143,113 @@
...
+          text: |
+            az network lb rule create -g MyResourceGroup --lb-name MyLb -n MyLbRule --protocol Tcp \\
+                --frontend-ip-name MyFrontEndIp --backend-pool-name MyAddressPool --backend-port 80 \\
+                --floating-ip true
+        - name: >
+            Create an HA ports load balancing rule that assigns a frontend IP and port to use all available backend IPs in a pool on the same port.
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 23]
```diff
@@ -1401,9 +2531,9 @@
           text: |
             az network nsg rule create -g MyResourceGroup --nsg-name MyNsg -n MyNsgRule --priority 4096 \\
                 --source-address-prefixes 208.130.28/24 --source-port-ranges 80 \\
-                --destination-address-prefixes '*' --destination-port-ranges 80 8080 --access Deny \\
+                --destination-address-prefixes `*` --destination-port-ranges 80 8080 --access Deny \\
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py (hunk 1 above)
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py (hunk 2 above)
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done. Removed "note", moved remaining to long-summary.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py (hunk 3 above)
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py (hunk 4 above)
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 24]
```diff
@@ -557,7 +557,7 @@
     type: command
     short-summary: Update a rule.
     examples:
-        - name: Update a rule use a new Http Listener.
+        - name: Update a rule use a new Http listener.
```
Comment (sptramer) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: Capitalize `HTTP`

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 25]
```diff
@@ -934,7 +935,9 @@
     short-summary: Add an A record.
     examples:
         - name: Add an A record.
-          text: az network dns record-set a add-record -g MyResourceGroup -z www.mysite.com -n MyRecordSet -a MyIpv4Address
+          text: |
+            az network dns record-set a add-record -g MyResourceGroup -z www.mysite.com\\
```
Comment (sptramer) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: Either put the example back on a single line (which is OK, shouldn't have called these out) or add whitespace before `\\` and indent all continuation lines.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 26]
```diff
@@ -2840,7 +2911,9 @@
         - name: Gets all the available BGP service communities.
           text: az network route-filter rule list-service-communities -o table
         - name: Get the community value for Exchange.
-          text: az network route-filter rule list-service-communities --query '[].bgpCommunities[?communityName==`Exchange`].[communityValue][][]' -o tsv
+          text: |
+            az network route-filter rule list-service-communities\\
+            --query '[].bgpCommunities[?communityName==`Exchange`].[communityValue][][]' -o tsv
```
Comment (sptramer) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: Indent continuation lines for all examples and put space before `\\` continuation

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py (hunk 24 above)
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: done

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py (hunk 25 above)
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: applied the latter in formatting

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py (hunk 26 above)
Comment (markjbrown) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: ok, went through all of the examples, found/fixed all examples with missing space before \\ and indentation. Cleaned up some others in there too.

**Comment by sptramer** [comment]
Looks good 👍

**Comment by markjbrown** [comment]
Whew awesome! Do you know when APEX will be publishing next? I'd love to get these out before EOM if possible. Thanks again for your help reviewing.

**Comment by sptramer** [comment]
I'm not sure when ref is picked up for automated builds by the system, but the publish to the docs site is 3PM daily. If there were recent changes to the reference I normally see them there, but I don't know if ref build is triggered by a merge to CLI or if it runs at a set time.

**Comment by tjprescott** [comment]
There are CI issues that need to be resolved before we can merge. It seems like your PR needs to be rebased to resolve the version issues, and the there's the linter (trailing whitespace, etc) issue. Finally, on a prior build I saw there were issues with non-ASCII characters being added to the help, which will break.

**Comment by markjbrown** [comment]
@tjprescott can I hop on a call with you tomorrow am to run through these issues? shouldn't take long.

**Comment by tjprescott** [comment]
@markjbrown sounds good

**Pull request merged**